src/reader/lexer.rs

```diff
@@ -221,7 +221,7 @@ pub struct Lexer {
     skip_errors: bool,
     inside_comment: bool,
     inside_token: bool,
-    eof_handled: bool
+    pub eof_handled: bool
```

Comment:
I would prefer having an mutator method like `fn reset_eof_handled(&mut self)` instead of making the field public, simply because other fields are not.